### PR TITLE
Backport fixes from develop to v6

### PR DESCRIPTION
**Pull request by jnyrup**
Since developing v7 is a larger undertaking than we initially expected, we want to backport _some fixes_ to v6.

This backporting should cover [PRs](https://github.com/fluentassertions/fluentassertions/pulls?q=is%3Apr+is%3Amerged+label%3Abug+-label%3A%22breaking+change%22+merged%3A2023-08-23..2024-08-18+sort%3Acreated-asc+) that were merged since v6.12 and marked as bugs.

🛑 Please do not ask for backporting new features - new features come with new bugs.

## IMPORTANT 

* [ ] If the PR touches the public API, the changes have been approved in a separate issue with the "api-approved" label.
* [ ] The code complies with the [Coding Guidelines for C#](https://www.csharpcodingguidelines.com/).
* [ ] The changes are covered by unit tests which follow the Arrange-Act-Assert syntax and the naming conventions such as is used [in these tests](../tree/develop/Tests/FluentAssertions.Equivalency.Specs/MemberMatchingSpecs.cs#L51-L430).
* [ ] If the PR adds a feature or fixes a bug, please update [the release notes](../tree/develop/docs/_pages/releases.md) with a functional description that explains what the change means to consumers of this library, which are published on the [website](https://fluentassertions.com/releases).
* [ ] If the PR changes the public API the changes needs to be included by running [AcceptApiChanges.ps1](../tree/develop/AcceptApiChanges.ps1) or [AcceptApiChanges.sh](../tree/develop/AcceptApiChanges.sh).
* [ ] If the PR affects [the documentation](../tree/develop/docs/_pages), please include your changes in this pull request so the documentation will appear on the [website](https://www.fluentassertions.com/introduction).
    * [ ] Please also run `./build.sh --target spellcheck` or `.\build.ps1 --target spellcheck` before pushing and check the good outcome

### CODE REVIEW

File: docs/_pages/releases.md [hunk 1]
```diff
@@ -7,6 +7,32 @@ sidebar:
   nav: "sidebar"
 ---
 
+## 6.13.0
```
Comment (IT-VBFK) [inline docs/_pages/releases.md:10]: Should this be 6.12.1?

Comment (jnyrup) [reply docs/_pages/releases.md]: Yes, thanks 👍

**Review by dennisdoomen — APPROVED** [review]
Great work. I do think this PR should address the Qodana issues (if they are genuine).

**Comment by ITaluone** [comment]
Missed the tag? 😉

**Pull request merged**